.github/PULL_REQUEST_TEMPLATE.md

```diff
@@ -0,0 +1,24 @@
+<!--
+Thank you for your pull request. Please provide a description above and
+review the checklist below.
+
+Contributors guide: ./CONTRIBUTING.md
+-->
+
+## Checklist
+<!--
+Remove items that do not apply. For completed items, change [ ] to [x].
+-->
+
+- [ ] Keep pull requests small so they can be easily reviewed.
+- [ ] Update the documentation.
+- [ ] Update tests.
+- [ ] Categorize the PR by setting a good title and adding one of the labels:
+      `bug`, `feature`, `ui`, `change`, `documentation`, `breaking`, `ci`
+      as they show up in the changelog
```

Comment:
Perhaps we could add a `\<!-- ... -->` block here with explanation of what we think is meant by `ui`, `change` etc? Or am I overthinking this?

Reply:
A few additional thoughts:

Perhaps we should clarify \- [ ] Update the documentation to contain sub checkboxes:
\- [ ] Documentation for users has been updated (e.g. the [readme](https://github.com/LCA-ActivityBrowser/activity-browser/blob/master/README.md) or [wiki](https://github.com/LCA-ActivityBrowser/activity-browser/wiki)
\- [ ] All functions and classes have docstrings following [PEP 257](https://peps.python.org/pep-0257/)
\- [ ] All functions have type hinting following [PEP 484](https://peps.python.org/pep-0484/)
This could help contributors to make sure their code is well written -even though we still need to do all of the above for most of AB-

We may also want an additional check:
\- [ ] All relevant issues/discussions/projects have been tagged in the description
This will help maintainers keep track of issues to close and progress in general. I think there could be quite a few issues still open that have long since been resolved because we simply missed them during a merge and then forgot about them.

Reply:
> Perhaps we could add a \<!-- ... --> block here with explanation of what we think is meant by ui, change etc? Or am I overthinking this?

The checkbox for setting the label is in my opinion more of a reminder for the reviewer to make sure that the PR shows up in the changelog, so the reviewer can always clarify or change the label if the person opening the PR doesn't know what label to use. In order not to overload the PR template I would leave it as it is.

> Perhaps we should clarify - [ ] Update the documentation to contain sub checkboxes:
> - [ ] For features: Documentation for users has been updated (e.g. the [readme](https://github.com/LCA-ActivityBrowser/activity-browser/blob/master/README.md) or [wiki](https://github.com/LCA-ActivityBrowser/activity-browser/wiki))
> - [ ] All functions and classes have docstrings following [PEP 257](https://peps.python.org/pep-0257/)
> - [ ] All functions have type hinting following [PEP 484](https://peps.python.org/pep-0484/)
> This could help contributors to make sure their code is well written -even though we still need to do all of the above for most of AB-

While I agree that this would be nice to have, I'm afraid this could potentially scare away contributors. I guess that's something we could add once the AB code base is in a state where we can realistically expect the same for no contributions. But asking for type hints when there's not even an example in the AB code base how to do that is a bit much to ask imho.

Also we can always ask for improvements in the review, I wouldn't make any of this a hard requirement in the PR template.

> We may also want an additional check:
> - [ ] All relevant issues/discussions/projects have been tagged in the description
> This will help maintainers keep track of issues to close and progress in general. I think there could be quite a few issues still open that have long since been resolved because we simply missed them during a merge and then forgot about them.

There is already `- [ ] Link this PR to related issues`, is that not clear enough?

Reply:
Regarding the first point; do you think simply adding in brackets what the abbreviations (ui, ci etc.) mean would already be enough?

Reply:
Alright, I think you're right, this may indeed be too much. I got a bit carried away there. Lets keep it simple and we can indeed always ask for improvements during review.

>There is already `- [ ] Link this PR to related issues`, is that not clear enough?

It is, yes, I missed that.